tools/gen_dir_structure.py

```diff
@@ -0,0 +1,311 @@
+
+    consensus_path = dir_document / FILENAME_CONSENSUS
+    churn_path = dir_document / FILENAME_CHURN
+
+    generate_churninfo(consensus_path, churn_path, consensus_latest)
+
+
+def update_dir(dir_auth: Path, root: Path, date_utc: datetime) -> None:
+    """
+    Update the info about the network's status.
+
+    :param dir_auth: directory containing info about the directory authority
+    :param root: directory where to put all the network status information
+    :param date_utc: current date and time in UTC
+    """
+    dir_current = root / date_utc.strftime("%Y%m%d")
+
+    if not dir_current.exists():
+        # create dir and generate custom consensus.
+        create_current_dir(dir_auth, dir_current)
+    else:
+        consensus = fetch_valid_consensus()
+        update_churn(dir_current, consensus)
+
+    potential_documents_dirs = list()
+
+    for dt_days in range(1, CONSENSUS_VALIDITY_DAYS):
+        dir_documents_name: str = (date_utc - timedelta(days=dt_days)).strftime("%Y%m%d")
+        potential_documents_dirs.append(dir_documents_name)
+
+    for dir_documents_name in potential_documents_dirs:
+        dir_documents = root / dir_documents_name
+        # Update churn
+        if dir_documents.is_dir():
+            if os.access(dir_documents, os.R_OK | os.W_OK):
+                consensus = fetch_valid_consensus()
+                update_churn(dir_documents, consensus)
+            else:
+                raise InconsistentDirectoryStructure(f"Directory {dir_documents} is inconsistent.")
+
+
+def directory_structure(namespace: Namespace) -> None:
+    """
+    Function to build or update a directory structure containing up-to-date network info.
+
+    :param namespace: namespace containing the parsed arguments.
+    """
+    date_utc = datetime.utcnow()
+    dir_auth: Path = namespace.authority_directory
+    dir_root: Path = namespace.root_directory
+
+    ensure_dir_rw_access(dir_root)
+    ensure_valid_authority_dir(dir_auth, date_utc)
+
+    update_dir(dir_auth, dir_root, date_utc)
+
+
+def main(program: str, arguments: List[str]) -> None:
+    """
+    Entrypoint of the program. Parse the arguments, and call the correct function.
+
+    :param program: name of the script.
+    :param arguments: arguments passed to the program.
+    """
+
+    parser = ArgumentParser(prog=program)
+    parser.add_argument(
+        "--authority-directory",
+        help="Directory containing files related to the Tor directory authority.",
+        type=Path,
+        default="authority"
+    )
+    parser.add_argument(
+        "--root-directory",
+        help="Directory containing up-to-date documents for using Lightarti-rest.",
+        type=Path,
+        default="documents-public"
+    )
+
+    namespace = parser.parse_args(arguments)
+
+    directory_structure(namespace)
+
+"""     try:
+        directory_structure(namespace)
+    except Exception as err:
+        LOGGER.error(err)
+        sys.exit(1)
+ """
```

Comment:
Maybe remove?

Reply:
Oops! Sorry, this is actually the correct code, I do want a log and not a stacktrace in production.